controllers/configurationpolicy_controller.go

```diff
@@ -571,6 +591,8 @@ func (r *ConfigurationPolicyReconciler) cleanUpChildObjects(plc policyv1.Configu
 					object.Properties.UID == uid {
 					needsDelete = true
 				}
+			} else { // None or not
+				needsDelete = false
```

Comment:
This is happening too late in the method since the code above is performing API queries only for the return values to never get used.

My suggestion is to add something to this effect in the start of the method:
```go
if !strings.EqualFold(string(plc.Spec.RemediationAction), "enforce") {
	return deletionFailures
}

if !(string(plc.Spec.PruneObjectBehavior) == "DeleteAll" || string(plc.Spec.PruneObjectBehavior) == "DeleteIfCreated") {
	return deletionFailures
}
```

Afterwards, you can also remove the if statement of `if strings.EqualFold(string(plc.Spec.RemediationAction), "enforce") {` on line 581 of the diff.